test/multi_process_test.rb

```diff
@@ -28,10 +28,89 @@
     Resque.schedule = schedules
     counts << Resque.schedule.size
 
-    threads.each(&:join)
+    counts += get_results_from_children(pids)
 
     counts.each_with_index do |c, i|
       assert_equal schedule_count, c, "schedule count is incorrect (c: #{i})"
     end
   end
+
+  test 'concurrent shutdowns and startups do not corrupt the schedule' do
+    counts = []
+    children = []
+
+    processes = 100
```

Comment:
This seems like a lot. Do we need this many in order to make the test fail consistently without the patch?